CMakeLists.txt

```diff
@@ -1,9 +1,12 @@
 cmake_minimum_required(VERSION 3.9)
 
-set(CMAKE_CXX_FLAGS -std=c++17)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
You should also add `https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD_REQUIRED.html`